lib/bundler/resolver/spec_group.rb

```diff
@@ -28,7 +29,7 @@ def to_specs
           lazy_spec = LazySpecification.new(name, version, s.platform, source)
           lazy_spec.dependencies.replace s.dependencies
           lazy_spec
-        end.compact
+        end.compact.uniq
```

Comment:
Don't return duplicated specs.
We can make `Bundler::Resolver` cleaner to remove duplicated specs here.

This needs a change in `LazySpecification`.